helix-term/src/commands.rs

```diff
@@ -2260,6 +2261,87 @@ fn buffer_picker(cx: &mut Context) {
     cx.push_layer(Box::new(overlayed(picker)));
 }
 
+fn jumplist_picker(cx: &mut Context) {
+    struct JumpMeta {
+        id: DocumentId,
+        path: Option<PathBuf>,
+        selection: Selection,
+        text: String,
+        is_current: bool,
+    }
+
+    impl ui::menu::Item for JumpMeta {
+        type Data = ();
+
+        fn label(&self, _data: &Self::Data) -> Spans {
+            let path = self
+                .path
+                .as_deref()
+                .map(helix_core::path::get_relative_path);
+            let path = match path.as_deref().and_then(Path::to_str) {
+                Some(path) => path,
+                None => SCRATCH_BUFFER_NAME,
```

Comment:
I’m not sure `[scratch]` is a good fallback if path is (partially) not valid UTF-8. Did you consider [to_string_lossy](https://doc.rust-lang.org/std/path/struct.Path.html#method.to_string_lossy) instead? This does not incur allocation when data is entirely valid UTF-8.

Reply:
Thanks for the view, Learned from your suggestions. Regarding this comment, I copied the logic from buffer picker, I am not sure if we shall use to_string_lossy, I think we can change it in both places if this is better.

CC: @cossonleo , @the-mikedavis

Reply:
I think that for the buffer it makes perfect sense to fall-back to [scratch] whereas here it seems an alternative is needed, of which to_string_lossy is one of.